inference-engine/src/preprocessing/ie_preprocess_data.hpp

```diff
@@ -58,9 +58,12 @@ class IPreProcessData : public details::IRelease {
 
     //FIXME: rename to verifyAplicable
     virtual void isApplicable(const Blob::Ptr &src, const Blob::Ptr &dst) = 0;
+
+protected:
+    ~IPreProcessData() = default;
```

Comment:
why is it protected and not virtual?

Reply:
As this and all others  entities in IE are loaded using `std::shared_ptr` (as they are created inside dynamic library and we should provide deleter from library too) we do not use deletion by base reference.  We protect from base destructor call from external code. Virtual just not needed here.